src/login/connect_server.h

```diff
@@ -0,0 +1,2257 @@
+﻿/*
+===========================================================================
+
+Copyright (c) 2023 LandSandBoat Dev Teams
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see http://www.gnu.org/licenses/
+
+===========================================================================
+*/
+
+#pragma once
+
+#include "common/application.h"
+#include "common/console_service.h"
+#include "common/logging.h"
+#include "common/lua.h"
+#include "common/md52.h"
+#include "common/settings.h"
+#include "common/sql.h"
+#include "common/utils.h"
+#include "map/packets/basic.h"
+
+// openssl applink.c prevents issues with debug vs release vs threaded/single threaded .dlls at runtime
+// apparently not an issue with linux
+#ifdef _WIN32
+#include <ms/applink.c>
+#endif
+
+#include <asio/ssl.hpp>
+#include <asio/ts/buffer.hpp>
+#include <asio/ts/internet.hpp>
+#include <chrono>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <thread>
+#include <type_traits>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+#include <zmq.hpp>
+
+#include <nonstd/jthread.hpp>
+
+#ifndef _WIN32
+#include <sys/resource.h>
+#endif
+
+// TODO: Move to enum
+#define LOGIN_ATTEMPT         0x10
+#define LOGIN_CREATE          0x20
+#define LOGIN_CHANGE_PASSWORD 0x30
+
+/*return result*/
+#define LOGIN_FAIL                    0x00
+#define LOGIN_SUCCESS                 0x01
+#define LOGIN_SUCCESS_CREATE          0x03
+#define LOGIN_SUCCESS_CHANGE_PASSWORD 0x06
+
+#define LOGIN_REQUEST_NEW_PASSWORD 0x05
+
+#define LOGIN_ERROR                     0x02
+#define LOGIN_ERROR_CREATE              0x09
+#define LOGIN_ERROR_CREATE_TAKEN        0x04
+#define LOGIN_ERROR_CREATE_DISABLED     0x08
+#define LOGIN_ERROR_CHANGE_PASSWORD     0x07
+#define LOGIN_ERROR_ALREADY_LOGGED_IN   0x0A
+#define LOGIN_ERROR_VERSION_UNSUPPORTED 0x0B
+
+#define SUPPORTED_XILOADER_VERSION "1.0.0"
+
+bool check_string(std::string const& str, std::size_t max_length)
+{
+    // clang-format off
+    return !str.empty() &&
+        str.size() <= max_length &&
+        std::all_of(str.cbegin(), str.cend(),
+        [](char const& c)
+        {
+            return c >= 0x20;
+        });
+    // clang-format on
+}
+
+class handler_session
```

Comment:
This is all going to need to be split out into a cpp file 💀

Reply:
I split it out into .h files, is that enough? please review